### PR TITLE
Ecc 1331 - Migrate to hmrc-mongo

### PR DESCRIPTION
**Pull request by beckfordp**
https://jira.tools.tax.service.gov.uk/browse/ECC-1331

### CODE REVIEW

File: app/uk/gov/hmrc/customs/rosmfrontend/services/cache/sessionCache.scala [hunk 1]
```diff
@@ -136,94 +119,102 @@ class SessionCache @Inject()(appConfig: AppConfig, mongo: ReactiveMongoComponent
...
+  def email(implicit request: Request[_]): Future[String] =
+    getData[String](emailKey).map(_.getOrElse(throwException(emailKey)))
 
-  def registerWithEoriAndIdResponse(implicit hc: HeaderCarrier): Future[RegisterWithEoriAndIdResponse] =
-    getCached[RegisterWithEoriAndIdResponse](
-      sessionId,
-      (cachedData, id) => cachedData.registerWithEoriAndIdResponse(id)
-    )
+  def hasNino(implicit request: Request[_]): Future[Option[Boolean]] =
+    getData[Boolean](hasNinoKey) // .map(_.getOrElse(throwException(hasNinoKey)))
```
Comment (darbyo) [inline app/uk/gov/hmrc/customs/rosmfrontend/services/cache/sessionCache.scala:156]: Is it intended behaviour that we dont map this value here?

File: app/uk/gov/hmrc/customs/rosmfrontend/services/cache/sessionCache.scala [hunk 2]
```diff
@@ -136,94 +119,102 @@ class SessionCache @Inject()(appConfig: AppConfig, mongo: ReactiveMongoComponent
...
 
-  def subscriptionStatusOutcome(implicit hc: HeaderCarrier): Future[SubscriptionStatusOutcome] =
-    getCached[SubscriptionStatusOutcome](sessionId, (cachedData, id) => cachedData.subscriptionStatusOutcome(id))
+  def mayBeEmail(implicit request: Request[_]): Future[Option[String]] =
+    getData[String](emailKey)
```
Comment (darbyo) [inline app/uk/gov/hmrc/customs/rosmfrontend/services/cache/sessionCache.scala:159]: As per previous comment

File: build.sbt [hunk 3]
```diff
@@ -119,26 +119,31 @@ val compileDependencies = Seq(
...
+  "org.scalatestplus"      %% "scalacheck-1-15"     % "3.2.11.0"          % "test,it",
+  "com.vladsch.flexmark"    % "flexmark-all"        % "0.62.0"   % "test,it",
   "com.github.tomakehurst" % "wiremock-standalone" % "2.27.2" % "test, it"
     exclude ("org.apache.httpcomponents", "httpclient") exclude ("org.apache.httpcomponents", "httpcore"),
-  "org.seleniumhq.selenium" % "selenium-htmlunit-driver" % "2.52.0" % "test,it",
-  "org.scalacheck" %% "scalacheck" % "1.14.0" % "test,it",
-  "org.jsoup" % "jsoup" % "1.11.3" % "test,it",
-  "us.codecraft" % "xsoup" % "0.3.1" % "test,it",
-  "org.mockito" % "mockito-core" % "3.11.1" % "test,it",
+//  "org.seleniumhq.selenium" % "selenium-htmlunit-driver" % "2.52.0" % "test,it",
```
Comment (darbyo) [inline build.sbt:142]: We can remove this if its not needed.

**Review by darbyo — APPROVED** [review]

File: app/uk/gov/hmrc/customs/rosmfrontend/services/cache/sessionCache.scala (hunk 1 above)
Comment (beckfordp) [reply app/uk/gov/hmrc/customs/rosmfrontend/services/cache/sessionCache.scala:156]: Yes. the getData call returns an Option. Sometime we want to return an Option to the client. Other times we want to throw an exception when we always expect a value. I'll delete the commented out code.

File: app/uk/gov/hmrc/customs/rosmfrontend/services/cache/sessionCache.scala (hunk 2 above)
Comment (beckfordp) [reply app/uk/gov/hmrc/customs/rosmfrontend/services/cache/sessionCache.scala:159]: Same thing.

File: build.sbt (hunk 3 above)
Comment (beckfordp) [reply build.sbt:142]: Yep.

**Pull request merged**